TS/src/BenchmarkRunner.lf

```diff
@@ -0,0 +1,100 @@
+target TypeScript;
+
+/*
+ *
+ * Copyright (C) 2022 Hanyang University
+ *
+ * @author Wonseo Choi
+ */
+
+reactor BenchmarkRunner(num_iterations:number(12)) {
+
+    output start:boolean;
+    input finish:boolean;
+    
+    logical action nextIteration:boolean;
+    logical action done:boolean;
+    
+    state count:number(0);
+    
+    /** Start time for runtime measurement. */
+    state startTime:time;
+    
+    /** Runtime measurements. */
+    state measuredTimes: {=Array<number>=}({= [] =});
+    
+    
+    reaction(startup) -> nextIteration {=
+        // Initialize an array of interval_t
+        measuredTimes.length = num_iterations;
+        actions.nextIteration.schedule(0, true);
+        console.log("Benchmark will run " + num_iterations + " times \n");
+    =}
+   
+    
+    reaction(nextIteration) -> start, done {=
+        if (count < num_iterations) { 
+            startTime = util.getCurrentPhysicalTime();
+            start = true;
+        } else {
+            actions.done.schedule(0, true);
+        }
+    =}
+    
+    reaction(finish) -> nextIteration {=
+        const duration = util.getCurrentPhysicalTime().subtract(startTime as TimeValue);
+        measuredTimes[count] = duration.toMilliseconds();
+        ++count;
+        
+        console.log("Iteration "+ count + " - " + duration.toMilliseconds() + " ms\n");
+        
+        actions.nextIteration.schedule(0, true);
+        
+    =}
+    
+    reaction(done) {=        
+        measuredTimes.sort()
+        
+        console.log("Execution - Summary:\n");
+        console.log("Best Time:\t " + measuredTimes[0] + " msec\n");
+        console.log("Worst Time:\t " + measuredTimes[num_iterations - 1] + " msec\n");
+        console.log("Median Time:\t " + median(measuredTimes, num_iterations) + " msec\n");
+        
+        util.requestStop();
+    =}
+    
+    preamble {=
+        function median(execTimes: Array<number>, size: number): number {
+            if (size == 0) {
+                return 0.0;
+            }
+
+            const middle = size / 2;
+
+            if(size % 2 == 1){
+                return execTimes[middle]
+            } else {
+                return (execTimes[middle-1] + execTimes[middle]) / 2;
+            }
+        }
+    =}
+
+    // TODO : Implement OS infrom !! // 
+    // preamble {=    
```

Comment:
Why are these comments checked in, and what is the code meant to accomplish?

Reply:
In other languages (C, Cpp)'s Benchmark runner, code determines the execution environment.
So, I tried about it but have not accomplished it yet.

Reply:
Oh, I wasn't aware of this. In any case I suggest removing the comment out code and leaving a `FIXME`. Here's how to do it on NodeJS: https://stackoverflow.com/questions/8683895/how-do-i-determine-the-current-operating-system-with-node-js

Reply:
Your comments have been reflected in f2140dc1dc9e7e18009dea80eb5e19844079e8da